lib/dgram.js

```diff
@@ -78,7 +78,7 @@ exports._createSocketHandle = function(address, port, addressType, fd, flags) {
 function Socket(type, listener) {
   EventEmitter.call(this);
 
-  if (typeof type === 'object') {
+  if (type && typeof type === 'object') {
```

Comment:
It actually is if the proper error message is to be thrown. Otherwise it fails with `null` with a "cannot access property type of null" message.